src/core/control/PdfCache.cpp

```diff
@@ -98,30 +101,40 @@ void PdfCache::render(cairo_t* cr, const XojPdfPageSPtr& popplerPage, double zoo
     if (needsRefresh) {
         double renderZoom = std::max(zoom, 1.0);
 
+        auto popplerPage = cacheResult ? cacheResult->popplerPage : pdfDocument.getPage(pageNo);
+
+        if (!popplerPage) {
+            g_error("PdfCache::render Could not get the pdf page %zu from the document", pageNo);
```

Comment:
nitpick: `g_warning` would probably be better here, since this is not a fatal error.